ConsumerApi.Tests.Integration/StepDefinitions/DevicesStepDefinitions.cs

```diff
@@ -75,12 +104,24 @@ public void GivenTheCurrentUserUsesD1()
     [Given(@"an un-onboarded device d2")]
     public async Task GivenAnUnOnboardedDeviceD2()
     {
-        var deviceResponse = await RegisterDevice(_createChallengeResponse!.Content.Result, _keyPair);
+        var deviceResponse = await RegisterDevice(_createChallengeResponse1!.Content.Result, _keyPair1);
         _deviceIdD2 = deviceResponse.Content.Result!.Id;
```

Comment:
You should create a new challenge instead of reusing the old one. Then you also don't need to save `_createChallengeResponse2` in a field